packages/odyssey-react-mui/src/Checkbox.tsx

```diff
@@ -131,9 +133,24 @@ const Checkbox = ({
 
   const onChange = useCallback<NonNullable<MuiCheckboxProps["onChange"]>>(
     (event, checked) => {
+      if (isReadOnly) {
```

Comment:
Same comment here, I think this being handled in the `onClick` is enough. If we prevent the change from happening in the first place, don't think it matters if this is readonly or not

Reply:
Good call, radio functions and has a different keyboard workflow, so I think this can be removed (still functions as expected when I tested).